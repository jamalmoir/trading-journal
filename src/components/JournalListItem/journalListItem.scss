.journal-list-item {
  padding: 0.5rem;
  display: flex;
  align-items: center;
}

.journal-list-item-kind {
  margin-left: 1rem;
}

.journal-list-item-edit {
  margin-left: auto;

  &:hover {
    cursor: pointer;
  }
}