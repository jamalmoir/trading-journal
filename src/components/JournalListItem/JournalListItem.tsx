import React from 'react';
import { Link } from 'react-router-dom';
import Types from 'Types';
import { JournalEditModal } from '../JournalEditModal';
import './journalListItem.scss';


interface JournalListProps {
  className?: string;
  journal: Types.Journal;
}

export const JournalListItem = (props: JournalListProps) => {
  let getBadgeType = () => {
    if (props.journal.kind === 'live') return 'badge-success';
    if (props.journal.kind === 'demo') return 'badge-primary';

    return 'badge-secondary';
  }

  return (
    <Link to={ 'journal/' + props.journal.id } className={ 'journal-list-item ' + props.className }>
      { props.journal.name } { /*| { props.journal.tradeCount } { props.journal.tradeCount === 1 ? ' trade ' : ' trades ' } */}
      <span className={ "journal-list-item-kind badge badle-pill " + getBadgeType() }>{ props.journal.kind }</span>
      <i className="journal-list-item-edit far fa-edit" data-toggle="modal" data-target="#journalEditModal"></i>
      <JournalEditModal id='journalEditModal' journal={ props.journal } />
    </Link>
  )
}