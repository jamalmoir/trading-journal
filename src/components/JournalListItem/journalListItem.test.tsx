
import * as React from 'react';
import { JournalListItem } from './JournalListItem';
import { shallow } from 'enzyme';
import Types from 'Types';


const testJournal: () => Types.Journal = () => ({
		id: 'someId',
		userId: 'someUserId',
		kind: 'live',
		currency: 'GBP',
		name: 'Some Journal',
		created: new Date('2019/09/10'),
		modified: new Date('2019/09/10'),
		tradeCount: 10,
});

describe("<JournalListItem />", () => {
  it('should match snapshot', () => {
    const component = shallow(<JournalListItem journal={ testJournal() }/>);
    expect(component).toMatchSnapshot();
	});
	
	it('should have the right badge depending on journal type', () => {
		const journal: Types.Journal = testJournal();
		const component = shallow(<JournalListItem journal={ journal }/>);
		
    component.setProps({journal: {...component.props().journal, kind: 'live'}});
    expect(component.find('span').hasClass('badge-success')).toEqual(true);
    expect(component.find('span').hasClass('badge-primary')).toEqual(false);
		expect(component.find('span').hasClass('badge-secondary')).toEqual(false);

    component.setProps({journal: {...component.props().journal, kind: 'demo'}});
    expect(component.find('span').hasClass('badge-success')).toEqual(false);
    expect(component.find('span').hasClass('badge-primary')).toEqual(true);
		expect(component.find('span').hasClass('badge-secondary')).toEqual(false);

    component.setProps({journal: {...component.props().journal, kind: 'backtest'}});
    expect(component.find('span').hasClass('badge-success')).toEqual(false);
    expect(component.find('span').hasClass('badge-primary')).toEqual(false);
    expect(component.find('span').hasClass('badge-secondary')).toEqual(true);
	})
});