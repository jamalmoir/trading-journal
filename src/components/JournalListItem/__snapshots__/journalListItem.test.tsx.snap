// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JournalListItem /> matches snapshot 1`] = `
<Link
  className="journal-list-item undefined"
  to="journal/someId"
>
  Some Journal
  <span
    className="journal-list-item-kind badge badle-pill badge-success"
  >
    live
  </span>
  <i
    className="journal-list-item-edit far fa-edit"
  />
</Link>
`;
